src/core/public/analytics/analytics_service.test.mocks.ts

```diff
@@ -0,0 +1,24 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { AnalyticsClient } from '@kbn/analytics-client';
+import { Subject } from 'rxjs';
+
+export const analyticsClientMock: jest.Mocked<AnalyticsClient> = {
+  optIn: jest.fn(),
+  reportEvent: jest.fn(),
+  registerEventType: jest.fn(),
+  registerContextProvider: jest.fn(),
+  removeContextProvider: jest.fn(),
+  registerShipper: jest.fn(),
+  telemetryCounter$: new Subject(),
+};
```

Comment:
> Can this be imported from `@kbn/analytics-client` using `jest.requireActual` somehow? 

I think it would make it harder to test because the exported type is an observable, so we won't be able to trigger test counters via `.next()`.

> I see this is duplicated from `packages/analytics/client/src/analytics_client/mocks.ts`

I wish I could simply import `@kbn/analytics-client/mocks` to avoid duplication of the mocks on the client and the core service. But I'm not able to make it work with the new Bazel package builds. And splitting the package like in the `@kbn/logging-mocks` feels like it would have the same issues.

Reply:
Okay, let's keep it that way then